lib/spack/spack/build_systems/autotools.py

```diff
@@ -123,10 +123,10 @@ def _do_patch_config_files(self):
                 # Then search in all sub directories recursively.
                 # We would like to use AC_CONFIG_AUX_DIR, but not all packages
                 # ship with their configure.in or configure.ac.
-                config_path = next((os.path.join(r, f)
+                config_path = next((os.path.abspath(os.path.join(r, f))
```

Comment:
This change appears to do the same thing (computing the abspath on the result seems the same as computing them in the generator). Is this depending on some behavior of `abspath`?

In other words: what issue does this resolve?

Reply:
Ah, sorry, should have mentioned this: If there is no config.guess, `next(..., None)` will return `None` and `abspath` will produce an error.

Reply:
@scheibelp I mentioned this on slack packages, but I didn't repeat it here.
Previously, `spack install papi` on powerpc and arm died with:

```
==> Error: TypeError: expected str, bytes or os.PathLike object, not NoneType
/home/krentel/liberty/spack-repo/lib/spack/spack/build_systems/autotools.py:128,
in _do_patch_config_files:
        125             # ship with their configure.in or configure.ac.
        126             config_path = next((os.path.join(r, f)
        127                                 for r, ds, fs in os.walk('.') for f in fs
  >>    128                                 if f == config_file), None)
        129             my_config_files[config_name] = os.path.abspath(config_path)
```